File: JavaR2/src/Solution0088.java
/**
 * <pre>
 * Modify Information:
 * Author       Date          Description
 * ============ ============= ============================
 * maximus         2024/2/20      create
 */
public class Solution0088 {
	//从末尾开始,从大到小复制
	public void merge(int[] nums1, int m, int[] nums2, int n) {
		int res = m + n - 1;
		m--;
		n--;

		while (m >= 0 && n >= 0) {
			if (nums1[m] >= nums2[n]) {
				nums1[res] = nums1[m];
				m--;
			} else if (nums1[m] < nums2[n]) {
				nums1[res] = nums2[n];
				n--;
			}
			res--;
		}

		while (n >= 0) {
			nums1[res] = nums2[n];
			n--;
			res--;
		}

	}
}

File: goR3/solution0088.go
package main

func Merge(nums1 []int, m int, nums2 []int, n int) {
	index := m + n - 1

	m, n = m-1, n-1

	for m >= 0 && n >= 0 {
		if nums1[m] > nums2[n] {
			nums1[index] = nums1[m]
			m--
		} else {
			nums1[index] = nums2[n]
			n--
		}
		index--
	}

	for m >= 0 {
		nums1[index] = nums1[m]
		m--
		index--
	}

	for n >= 0 {
		nums1[index] = nums2[n]
		n--
		index--
	}

}
